mod/b_extended_profile/actions/b_extended_profile/edit_profile.php

```diff
@@ -524,34 +483,71 @@
                     }
                 }
             }
-
             if ($user->portfolio == NULL) {
                 $user->portfolio = $portfolio_list_guids;
-
-            }else {
+            }
+            else {
                 $stack = $user->portfolio;
                 if (!(is_array($stack))) { $stack = array($stack); }
-
                 if ($portfolio_list_guids != NULL) {
                     $user->portfolio = array_merge($stack, $portfolio_list_guids);
                 }
             }
+            //$user->portfolio = null;
             $user->portfolio_access = $access;
             $user->save();
+            break;
+
+        /*
+         * MODIFIED CODE!
+         * optset is the array of opt-in choices and gets directly saved as a metadata array (cannot be associative).
+         * access is not currently being used for anything.
+         */
+        case 'opt-in':
+            $opt_in_set = get_input('opt_in_set');
+            $access = get_input('access');
+
+            $user->opt_in_missions = $opt_in_set[0];
+            $user->opt_in_swap = $opt_in_set[1];
+            $user->opt_in_mentored = $opt_in_set[2];
+            $user->opt_in_mentoring = $opt_in_set[3];
+            $user->opt_in_shadowed = $opt_in_set[4];
+            $user->opt_in_shadowing = $opt_in_set[5];
+            $user->opt_in_jobshare = $opt_in_set[6];
+            $user->opt_in_pcSeek = $opt_in_set[7];
+            $user->opt_in_pcCreate = $opt_in_set[8];
+            $user->opt_in_ssSeek = $opt_in_set[9];
+            $user->opt_in_ssCreate = $opt_in_set[10];
+            $user->opt_in_rotation = $opt_in_set[11];
+            $user->opt_in_assignSeek = $opt_in_set[12];
+            $user->opt_in_assignCreate = $opt_in_set[13];
+            $user->opt_in_deploySeek = $opt_in_set[14];
+            $user->opt_in_deployCreate = $opt_in_set[15];
+            $user->opt_in_missionCreate = $opt_in_set[16];
+            /*$user->opt_in_peer_coached = $opt_in_set[6];
+              $user->opt_in_peer_coaching = $opt_in_set[7];
+              $user->opt_in_skill_sharing = $opt_in_set[8];
+              $user->opt_in_job_sharing = $opt_in_set[9];*/
+
+            // Not saving this at the moment because it is not in use.
+            //$user->optaccess = $access;
+
+            $user->save();
 
             break;
+        /*
+         * END MODIFIED CODE
+         */
 
         default:
             system_message(elgg_echo("profile:saved"));
     }
-
+    //system_message(elgg_echo("profile:saved"));
     if($error == true){
        register_error(elgg_echo('Not all information could be saved, empty fields are not allowed'));
     } else {
         system_message(elgg_echo("profile:saved"));
     }
 }
-
-// In case this view will be called via the elgg_view_form() action, then we know it's the basic profile only
-else {
+else {  // In case this view will be called via the elgg_view_form() action, then we know it's the basic profile only
```

Comment:
You've move the comments in a bunch of other places which is making the diff worse